cli/src/repl.rs

```diff
@@ -1,59 +1,49 @@
-use crate::report::{print_flush, FormattedError};
-use context::source::Source;
-use dbg_pls::color;
-use miette::GraphicalReportHandler;
+use crate::cli::handle_source;
+use crate::report::print_flush;
+use context::source::{OwnedSource, Source};
 use parser::parse;
 use std::io;
 use std::io::BufRead;
 use std::io::Write;
 
-pub fn prompt(handler: GraphicalReportHandler) -> io::Result<()> {
+/// Indefinitely prompts a new expression to the stdin,
+/// displaying back the errors if any and the formed AST
+pub fn prompt() {
+    loop {
+        if let Some(source) = parse_input() {
+            handle_source(source.as_source());
+        }
+    }
```

Comment:
```suggestion
    while let Some(source) = parse_input() {
        handle_source(source.as_source());
    }
```
The loop must end somewhere where stdin ends.